tests/kernel/wave/wave_e2e_test.py

```diff
@@ -1623,3 +1624,59 @@ def test(
     test(a, scalar_c, scalar_d, b)
 
     assert torch.all(b == arg_vals[3]).item()
+
+
+@require_e2e
+@pytest.mark.parametrize("shape", [(16,16)])
+@param_bool("use_buffer_ops", "buf_ops")
+def test_atomic_min(shape, use_buffer_ops, request):
+    run_bench = request.config.getoption("--runperf")
+
+    M = tkl.sym.M
+    N = tkl.sym.N
+    ADDRESS_SPACE = tkl.sym.ADDRESS_SPACE
+
+    wave_size = 64
+    BLOCK_M = 1
+    BLOCK_N = sympy.Max(sympy.Min(N, 256), wave_size)
+
+    constraints: list[tkw.Constraint] = [
+        tkw.HardwareConstraint(
+            threads_per_wave=wave_size,
+            waves_per_block=(1, 1, 1),
+            vector_shapes={M: BLOCK_M, N: BLOCK_N},
+        )
+    ]
+    constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 1)]
+    constraints += [tkw.WorkgroupConstraint(N, BLOCK_N, 0)]
+    constraints += [tkw.WaveConstraint(M, BLOCK_M)]
+    constraints += [tkw.WaveConstraint(N, BLOCK_N)]
+
+    @tkw.wave(constraints)
+    def test(
+        a: tkl.Memory[M, N, ADDRESS_SPACE, tkl.i32],
+    ):
+        res = tkw.read(a)
+        neg_shmem = tkw.allocate(shape=(M,N), distributed_shape=(M,N), dtype=tkl.i32, address_space=tkl.AddressSpace.SHARED_MEMORY)
```

Comment:
Can you also try with global memory (if it works)?

Reply:
I'll follow up in the next PR if that's alright, global memory probably needs some work on interface